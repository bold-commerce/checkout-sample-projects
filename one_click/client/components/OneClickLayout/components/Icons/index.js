export { default as TimesCircle } from './TimesCirlce'
export { default as ArrowRight } from './ArrowRight'
export { default as CaretRight } from './CaretRight'