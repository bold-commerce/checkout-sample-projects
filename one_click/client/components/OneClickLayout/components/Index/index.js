export { default as IndexPage } from "./IndexPage";
export { default as IndexGuest } from "./IndexGuest";