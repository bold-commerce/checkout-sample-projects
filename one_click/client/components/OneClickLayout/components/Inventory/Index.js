export { default as inventory } from './Inventory'
export { default as InventoryItem } from './InventoryItem'