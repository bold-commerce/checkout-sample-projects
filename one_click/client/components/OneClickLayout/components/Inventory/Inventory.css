.Inventory__Main {
    padding: 16px 32px;
}

.Inventory__Message {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 16px;
    position: static;
    border: 1px solid rgba(0,0, 0, 0.2);
    line-height: 20px;
    margin-top: 16px;
    font-family: SF Pro Display;
    font-style: normal;
}

.Inventory__Title {
    color:rgba(0, 0, 0, 0.9);
    margin-bottom: 16px;
    font-size: 18px;
}

.Inventory__Description {
    font-size: 14px;
    color: rgba(0, 0, 0, 0.65);

}

.SoldOut {
    color: #B4002B;
    border: 1px solid #B4002B;
    background-color: #FFEBF0;
    padding: 4px 8px;
}

.Arrow {
    margin: 0px 8px;
    color: rgba(0, 0, 0, 0.65);
    font-size: x-large;
}

.InventoryItem {
    display: flex;
}
.InventoryItem__ProductDetails {
    padding: 0 20px;
    width: 100%;
}

.InventoryItem__ProductQuantityWrapper {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    margin-top: 4px;
}

.InventoryItem__ProductImageWrapper {
    width: 72px;
    height: 72px;
    padding: 3.6px;
    position: relative;
}

.InventoryItem__RemoveItem {
    width: 16px;
    height: 16px;
    border: none;
    background-color: inherit;
    cursor: pointer;
}

.ReturnLink {
    display: inline-block;
    line-height: 20px;
    font-size: 14px;
    margin-top: 16px;
}