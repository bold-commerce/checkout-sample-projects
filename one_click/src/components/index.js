export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
