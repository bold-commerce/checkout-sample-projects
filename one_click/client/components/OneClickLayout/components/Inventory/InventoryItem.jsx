import React from "react";
import PropTypes from 'prop-types';
import { Details, Image } from "@boldcommerce/stacks-ui/lib";
import './Inventory.css'
import { ArrowRight, TimesCircle } from "../Icons";

const InventoryItem = ({
    title, description, orderQty, stockQty, onRemove, image
}) => (
    <div className="SummaryBlock InventoryItem" >
        <div className="InventoryItem__ProductImageWrapper">
            <Image
                title={title}
                alt={title}
                src={image}
            />
        </div>
        <div className="InventoryItem__ProductDetails">
            <Details
                title={title}
                description={description}
            />
            <div className="InventoryItem__ProductQuantityWrapper">
                {
                    stockQty === 0 ?
                    <div className="SoldOut">Sold Out</div> :
                    <>                        
                        <div aria-label="ordered quantity" className="ProductQuantity"> 
                            {orderQty} 
                        </div>
                        <div 
                            className="Arrow"
                        >
                            <ArrowRight />
                        </div>
                        <div aria-label="available quantity" className="ProductQuantity">
                            {stockQty}
                        </div>
                    </>
                }
            </div>
        </div>
        <div className="InventoryItem__RemoveItemWrapper">
            {stockQty !== 0 &&
            <button 
                aria-label="remove item"
                className="InventoryItem__RemoveItem" 
                onClick={onRemove}
            >
                <TimesCircle />
            </button>
            }
        </div>
    </div>
);

InventoryItem.propTypes ={
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    orderQty: PropTypes.number.isRequired,
    stockQty: PropTypes.number.isRequired,
    onRemove: PropTypes.func,
    image: PropTypes.string,
};

export default InventoryItem;