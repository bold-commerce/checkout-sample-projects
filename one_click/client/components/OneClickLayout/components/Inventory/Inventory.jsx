import React, { useState, useContext } from "react";
import InventoryItem from "./InventoryItem";
import { useHistory, useLocation } from "react-router";
import { Button } from "@boldcommerce/stacks-ui"
import { Link } from "react-router-dom";
import { useLineItems } from "@boldcommerce/checkout-react-components";
import { AppContext } from '../../context/AppContext';
import { Header } from "../Header";
import './Inventory.css';

export const Inventory = ({}, ref) => {
    const { data: lineItems, updateLineItemQuantity, removeLineItem } = useLineItems();
    const [loading, setLoading] = useState(false);
    const { websiteName } = useContext(AppContext);
    const history = useHistory();
    const location = useLocation();
    const inventory = location.state;

    let invItems = lineItems.map((item) => {
        const product = inventory.find(i => i.id === item.product_data.product_id);
        item.product_data.stock = product.stock;
        item.product_data.inventory_issue = item.product_data.quantity > item.product_data.stock;
        if (item.product_data.inventory_issue) {
            return (
                <InventoryItem
                    key={item.product_data.line_item_key} 
                    title={product.name}
                    description={item.product_data.title}
                    orderQty={item.product_data.quantity}
                    stockQty={item.product_data.stock}
                    image={item.product_data.image_url}
                    onRemove={() => removeLineItem(item.product_data.line_item_key) }
                />);
        }
    })

    async function handleUpdateQuantities() {
        setLoading(true);
        const results = [];

        for (let i = 0; i < lineItems.length; i++) {
            const item = lineItems[i];
            if ( item.product_data.inventory_issue ) {
                if ( item.product_data.stock > 0 ) {
                    results.push(updateLineItemQuantity(item.product_data.line_item_key, item.product_data.stock));
                } else {
                    results.push(removeLineItem(item.product_data.line_item_key));
                }
            }
        }

        await Promise.all(results);
        setLoading(false);
        history.push("/");
    }

    return (
        <div ref={ref} className="Inventory__Main">
            <Header title={websiteName} />
            <div className="Inventory__Message">
                <div className="Inventory__Title">
                    Inventory issues
                </div>                
                <div className="Inventory__Description">
                    Some Products became unavailable and your cart has been updated, We're sorry for the inconvenience.
                </div>
            </div>
            {invItems}
            <div className="Inventory__footer">
                <Link className="ReturnLink" to="/">Return to cart</Link>
                <Button
                    onClick={handleUpdateQuantities}
                    className="CheckoutButton"
                    disabled={loading}
                >
                    Continue with changes
                </Button>
            </div>
        </div>
    )
}

const InventoryForwardedRef = React.forwardRef(Inventory);

export default InventoryForwardedRef;