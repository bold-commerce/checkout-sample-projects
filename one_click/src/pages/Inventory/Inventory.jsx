import React, { useState, useContext } from "react";
import { InventoryItem } from "../../components/InventoryItem";
import { useNavigate, useLocation, Link } from "react-router-dom";
import { Button } from "@boldcommerce/stacks-ui"
import { useLineItems } from "@boldcommerce/checkout-react-components";
import { useVariants } from "../../hooks";
import { AppContext } from '../../context/AppContext';
import { Header } from "../../components/Header";
import './Inventory.css';
import { useTranslation } from 'react-i18next';

export const Inventory = ({}, ref) => {
    const { data: lineItems, updateLineItemQuantity, removeLineItem } = useLineItems();
    const [loading, setLoading] = useState(false);
    const { websiteName } = useContext(AppContext);
    const handleVariants = useVariants();
    const navigate = useNavigate();
    const location = useLocation();
    const inventory = location.state;
    const { t } = useTranslation();

    let invItems = lineItems.map((item) => {
        const variantId = item.product_data.variant_id;
        item.product_data.stock = inventory[variantId].quantity;
        item.product_data.inventory_issue = inventory[variantId].inventory_tracker !== 'none' && item.product_data.quantity > item.product_data.stock;
        if (item.product_data.inventory_issue) {
            return (
                <InventoryItem
                    key={item.product_data.line_item_key} 
                    title={item.product_data.product_title}
                    variants={handleVariants(item.product_data.title)}
                    orderQty={item.product_data.quantity}
                    stockQty={item.product_data.stock}
                    image={item.product_data.image_url}
                    onRemove={() => removeLineItem(item.product_data.line_item_key) }
                />);
        }
    })

    async function handleUpdateQuantities() {
        setLoading(true);
        const results = [];

        for (let i = 0; i < lineItems.length; i++) {
            const item = lineItems[i];
            if ( item.product_data.inventory_issue ) {
                if ( item.product_data.stock > 0 ) {
                    results.push(updateLineItemQuantity(item.product_data.line_item_key, item.product_data.stock));
                } else {
                    results.push(removeLineItem(item.product_data.line_item_key));
                }
            }
        }

        await Promise.all(results);
        setLoading(false);
        navigate("/");
    }

    return (
        <div ref={ref} className="Inventory__Main">
            <Header title={websiteName} />
            <div className="Inventory__Message">
                <div className="Inventory__Title">
                    {t('inventory.issues')}
                </div>                
                <div className="Inventory__Description">
                    {t('inventory.issues_description')}
                </div>
            </div>
            {invItems}
            <div className="Inventory__footer">
                <Link to="/" className="ReturnLink" >{t('inventory.return_to_cart')}</Link>
                <Button
                    onClick={handleUpdateQuantities}
                    className="CheckoutButton"
                    disabled={loading}
                >
                    {t('inventory.continue')}
                </Button>
            </div>
        </div>
    )
}

const InventoryForwardedRef = React.forwardRef(Inventory);

export default InventoryForwardedRef;