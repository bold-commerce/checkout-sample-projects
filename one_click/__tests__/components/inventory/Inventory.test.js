import React from 'react';
import { render } from '@testing-library/react';
import { Inventory } from '../../../client/components/OneClickLayout/components/Inventory';
import { MemoryRouter } from 'react-router';
import { 
    exampleLineItems as MOCKexampleLineItems,
    exampleInventory as MOCKinventory
} from '../../utils/lineItemHelpers';

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    useLineItems: () => ({
        data: MOCKexampleLineItems,
        updateLineItemQuantity: (() => {}),
        removeLineItems: (() => {})
    })
})).mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useLocation: () => ({
        state: MOCKinventory
    })
}));

describe('Inventory', () => {
    test('renders Inventory component', () => {
        const { asFragment } = render( 
            <Inventory />,
            {wrapper: MemoryRouter}
        );
        expect(asFragment()).toMatchSnapshot();
    })
})