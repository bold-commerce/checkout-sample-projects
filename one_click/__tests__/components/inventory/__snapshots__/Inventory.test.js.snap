// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory renders Inventory component 1`] = `
<DocumentFragment>
  <div
    class="Inventory__Main"
  >
    <div
      class="Inventory__Message"
    >
      <div
        class="Inventory__Title"
      >
        Inventory issues
      </div>
      <div
        class="Inventory__Description"
      >
        Some Products became unavailable and your cart has been updated, We're sorry for the inconvenience.
      </div>
    </div>
    <div
      class="SummaryBlock InventoryItem"
    >
      <div
        class="InventoryItem__ProductImageWrapper"
      >
        <div
          class="stx-image"
        >
          <div
            class="stx-image--content"
          >
            <div
              class="stx-image--holder"
            >
              <img
                alt="Test title 2, Revenge of the title"
                class="stx-image--image"
                src=""
                title="Test title 2, Revenge of the title"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="InventoryItem__ProductDetails"
      >
        <div
          class="stx-details stx-details--left"
        >
          <div
            class="stx-details__title"
          >
            Test title 2, Revenge of the title
          </div>
          <div
            class="stx-details__description"
          >
            Test title 2, Revenge of the title
          </div>
        </div>
        <div
          class="InventoryItem__ProductQuantityWrapper"
        >
          <div
            class="ProductQuantity"
          >
            2
          </div>
          <div
            class="Arrow"
          >
            â†’
          </div>
          <div
            class="ProductQuantity"
          >
            1
          </div>
        </div>
      </div>
      <div
        class="InventoryItem__RemoveItemWrapper"
      >
        <button
          class="InventoryItem__RemoveItem"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="SummaryBlock InventoryItem"
    >
      <div
        class="InventoryItem__ProductImageWrapper"
      >
        <div
          class="stx-image"
        >
          <div
            class="stx-image--content"
          >
            <div
              class="stx-image--holder"
            >
              <img
                alt="Test title 3, the search for more money"
                class="stx-image--image"
                src=""
                title="Test title 3, the search for more money"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="InventoryItem__ProductDetails"
      >
        <div
          class="stx-details stx-details--left"
        >
          <div
            class="stx-details__title"
          >
            Test title 3, the search for more money
          </div>
          <div
            class="stx-details__description"
          >
            Test title 3, the search for more money
          </div>
        </div>
        <div
          class="InventoryItem__ProductQuantityWrapper"
        >
          <div
            class="SoldOut"
          >
            Sold Out
          </div>
        </div>
      </div>
      <div
        class="InventoryItem__RemoveItemWrapper"
      />
    </div>
    <div
      class="Inventory__footer"
    >
      <a
        class="ReturnLink"
        href="/"
      >
        Return to cart
      </a>
      <button
        aria-busy="false"
        aria-disabled="false"
        class="stx-button CheckoutButton"
      >
        <span
          class="stx-button__text"
        >
          Continue with changes
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
